core/controllers/base.py

```diff
@@ -635,3 +635,23 @@ def get(self):
         self.render_json({
             'token': csrf_token,
         })
+
+
+class OppiaMLVMHandler(BaseHandler):
+    """Base class for the handlers that communicate with Oppia-ML VM instances.
+    """
+
+    def extract_request_message_vm_id_and_signature(self):
+        """Returns the message, vm_id and signature from the incoming request.
+
+        Since incoming request can be either a protobuf serialized binary or
+        a JSON object, the derived classes must implement the necessary
+        logic to decode the incoming request and return a tuple of size 3
+        where message is at index 0, vm_id is at index 1 and signature is at
```

Comment:
Why not a domain object? More extensible if we need to modify fields in the future.

Reply:
This function is used to extract the fields relevant during the authentication of Oppia-ml which is very specific use case. Hence, the scope and usage of this is very specific where having a domain object would be an overkill for the task.

Reply:
I actually thought that at first, but as I read more through the PR I think I lean fairly strongly towards having a domain object. That's because you make use of this quite a bit and it appears in several method signatures where everything needs to be in quite a specific format.

Reply:
Done.